x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { ElasticsearchClient, SavedObjectsClientContract } from '@kbn/core/server';
+import {
+  AgentPolicyServiceInterface,
+  AgentService,
+  PackagePolicyServiceInterface,
+  PackageService,
+} from '@kbn/fleet-plugin/server';
+import { Installation, ListResult, PackagePolicy } from '@kbn/fleet-plugin/common';
+import {
+  CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+  INFO_ROUTE_PATH,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+} from '../../../common/constants';
+import { CspAppContext } from '../../plugin';
+import type { CspRouter } from '../../types';
+import { CspSetupStatus, Status } from '../../../common/types';
+import {
+  addRunningAgentToAgentPolicy,
+  getCspAgentPolicies,
+  getCspPackagePolicies,
+} from '../../lib/fleet_util';
+
+export const INDEX_TIMEOUT_IN_HOURS = 5;
+
+const isFindingsExists = async (esClient: ElasticsearchClient): Promise<boolean> => {
+  try {
+    const queryResult = await esClient.search({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+      query: {
+        match_all: {},
+      },
+      size: 1,
+    });
+
+    const hasLatestFinding = !!queryResult.hits.hits.length;
+
+    return hasLatestFinding ? true : false;
+  } catch (e) {
+    return false;
+  }
+};
+
+const getHealthyAgents = async (
+  soClient: SavedObjectsClientContract,
+  cspPackagePolicies: ListResult<PackagePolicy>,
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService
+): Promise<number> => {
+  const agentPolicies = await getCspAgentPolicies(
+    soClient,
+    cspPackagePolicies.items,
+    agentPolicyService
+  );
+  const enrichAgentPolicies = await addRunningAgentToAgentPolicy(agentService, agentPolicies);
+  const initialValue = 0;
+  const totalAgents = enrichAgentPolicies
+    .map((agentPolicy) => (agentPolicy.agents ? agentPolicy.agents : 0))
+    .reduce((previousValue, currentValue) => previousValue + currentValue, initialValue);
+
+  return totalAgents;
+};
+
+const getTimeDeltaSinceInstallation = (packageInfo: Installation): number => {
+  const installTime = packageInfo!.install_started_at;
+
+  // calculate time delta and convert to hours
+  const dt = (new Date().getTime() - new Date(installTime).getTime()) / (1000 * 60 * 60);
```

Comment:
- why do we need the `!` on `packageInfo` ? 

- we should probably make sure `installTime` is a valid argument to `new Date` so it doesn't accidently create a date we didn't mean to. (like `new Date(null)`)